cmd/vela-email/main.go

```diff
@@ -0,0 +1,284 @@
+			},
+			&cli.IntFlag{
+				EnvVars: []string{"VELA_BUILD_STARTED", "BUILD_STARTED"},
+				Name:    "build-started",
+				Usage:   "environment variable reference for reading in build started",
+			},
+		},
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		logrus.Fatal(err)
+	}
+}
+
+func run(c *cli.Context) error {
+	switch c.String("log.level") {
+	case "t", "trace":
+		logrus.SetLevel(logrus.TraceLevel)
+	case "d", "debug":
+		logrus.SetLevel(logrus.DebugLevel)
+	case "w", "warn":
+		logrus.SetLevel(logrus.WarnLevel)
+	case "e", "error":
+		logrus.SetLevel(logrus.ErrorLevel)
+	case "f", "fatal":
+		logrus.SetLevel(logrus.FatalLevel)
+	case "p", "panic":
+		logrus.SetLevel(logrus.PanicLevel)
+	case "i", "info":
+		fallthrough
+	default:
+		logrus.SetLevel(logrus.InfoLevel)
+	}
+
+	if c.IsSet("ci") {
+		logrus.SetFormatter(&logrus.TextFormatter{
+			DisableColors: true,
+			FullTimestamp: true,
+		})
+	} else {
+		logrus.SetFormatter(&logrus.TextFormatter{
+			ForceColors:   true,
+			FullTimestamp: false,
+			PadLevelText:  true,
+		})
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"code":     "https://github.com/go-vela/vela-email",
+		"docs":     "https://go-vela.github.io/docs/plugins/registry/email/",
+		"registry": "https://hub.docker.com/r/target/vela-email",
+	}).Info("Vela Email Plugin")
+
+	// create the plugin
+	p := &Plugin{
+		// sendType configuration
+		SendType: c.String("sendtype"),
+		// auth configuration
+		Auth: c.String("auth"),
+
+		// attachment configuration
+		Attachment: &email.Attachment{
+			Filename: c.String("filename"),
+		},
+
+		// email configuration
+		Email: &email.Email{
+			ReplyTo:     c.StringSlice("replyto"),
+			From:        c.String("from"),
+			To:          c.StringSlice("to"),
+			Bcc:         c.StringSlice("bcc"),
+			Cc:          c.StringSlice("cc"),
+			Subject:     c.String("subject"),
+			Text:        []byte(c.String("text")),
+			HTML:        []byte(c.String("html")),
+			Sender:      c.String("sender"),
+			ReadReceipt: c.StringSlice("readreceipt"),
+		},
+
+		// smtp configuration
+		SMTPHost: &SMTPHost{
+			Host:     c.String("host"),
+			Port:     c.String("port"),
+			Username: c.String("username"),
+			Password: c.String("password"),
+		},
+
+		// tls configuration
+		TLSConfig: &tls.Config{
+			ServerName:         c.String("host"),
+			InsecureSkipVerify: c.Bool("skipverify"), //nolint:gosec // ignore false positive
+		},
+
+		// User Friendly Build configuration
+		BuildEnv: &BuildEnv{
+			BuildCreated:  time.Unix(int64(c.Int("build-created")), 0).UTC().String(),
+			BuildEnqueued: time.Unix(int64(c.Int("build-enqueued")), 0).UTC().String(),
+			BuildFinished: time.Unix(int64(c.Int("build-finished")), 0).UTC().String(),
+			BuildStarted:  time.Unix(int64(c.Int("build-started")), 0).UTC().String(),
+		},
+	}
+
+	// validates the plugin
+	if err := p.Validate(); err != nil {
+		return err
+	}
+
+	// add vela environment variables
+	p.Env = p.Environment()
```

Comment:
This feels a little awkward to me. We could probably remove the `Env` field and anywhere the plugin was using that field we can use the `Environment()` method instead. I don't think there are many (just 1?) reference to this so it's small but it'd make things a little simpler.